### DISCUSSION TITLE
Skip lefthook entirely if no changes in folder

### DISCUSSION THREAD

**Discussion by stychu** [evilmartians/lefthook#408] [Q&A] [answered]
Hi, I just found about left hook and would like to use it in my company project.
We have a monorepo with different technologies. One folder is `app/` is intended for nodejs project on which I want to run commands before commiting a files.
I've a one problem tho. I want to run commands ONLY when any file in the `app/` folder is changed/added/removed.
The lefthook should not even run when there are any other changes in the repo but no changes within `app/` 

I managed to skipp the commands if not changes in `app/` but this should not skip but rather not even RUN the lefthook as it takes around 5-10 seconds just to finish the lefthook process which other guys dont want to wait for unnecessarily.

How can I prevent lefthook from running in such case? Providing LEFTHOOK=0 git commit ... manually is NOT an option. I need this to be automatic based on changes from `app/` folder

**Comment by mrexox** [comment, marked as answer]
Hey! Do I properly understand the issue: lefthook executes and skips the steps as expected, but its execution takes too long?

How is lefthook installed on your system? It feels that lefthook is executed with `npx` and that's why it takes so much time. If this is the issue, consider installing it globally with [any supported way](https://github.com/evilmartians/lefthook/blob/master/docs/install.md#install-lefthook). 

Is it added to `package.json` as an NPM package? In that case, could you provide the sample project structure? The main question is: where `node_modules` folder locates related to the git root? It might be that for some reason it doesn't find an executable in node_modules and moves to `npx` fallback.

Please, provide a `lefthook.yml` config, so I am sure I haven't missed anything.

The general answer to your question is the following: lefthook must read the config to skip, so it must be executed at least to skip execution. So, the problem is - it should not take so much time just to be executed and skip.

**Reply by stychu** [reply]
Yess indeed I installed lefthook as npm dependency which is located in subfolder to the root folder.
After Installing lefthook from brew the check is instant now when its skipped. Work as I expected now.

Thank you !
